app/template/rest.py

```diff
@@ -46,12 +48,24 @@ def _content_count_greater_than_limit(content, template_type):
     return template.content_count > SMS_CHAR_COUNT_LIMIT
 
 
+def validate_parent_folder(template_json):
+    if template_json.get("parent_folder_id"):
+        return TemplateFolder.query.filter_by(
```

Comment:
I've updated the dao method in https://github.com/alphagov/notifications-api/pull/2213 to filter based on `service_id`, so you could use that here